Aerolyzer/app/views.py

```diff
@@ -59,16 +62,87 @@ def gallery(request):
 
 @login_required
 def upload(request):
-    return render(request,
-    'app/upload.html',
-	{ 'user': request.user },
-    )
+	if request.method == 'POST' and request.FILES['myfile']:
+		myfile = request.FILES['myfile']
+		fs = FileSystemStorage()
+		filename = fs.save(myfile.name, myfile)
+		uploadedFileUrl = fs.url(filename)
+		request.session['filename'] = filename
+		request.session['uploadedFileUrl'] = uploadedFileUrl
+		# verified = imgRestFuncs(filename)
+		# if not verified['isVerified']:
+		# 	return render(request,
+		# 	'app/upload.html',
+		# 	{ 'user': request.user, 'filename': filename, 'uploadedFileUrl': uploadedFileUrl,
+		# 	'error_message': verified['data']},)
+		# else:
+		#	request.session['exifData'] = verified['data']
+		return HttpResponseRedirect('retrieve')
+	return render(request, 'app/upload.html', { 'user': request.user, },)
 
 @login_required
 def profile(request):
     return render(request,
     'app/profile.html',
-	{ 'user': request.user },
+	{ 'user': request.user, },
+    )
+
+@login_required
+def retrieve(request):
+	uploadedFileUrl = request.session['uploadedFileUrl']
+	filename = request.session['filename']
+	# exifData = request.session['exifData']
+
+	if request.method == 'POST':
+		# wunderData = retrieve_weather_info(exifData['location'])
+		# if wunderData is None:
+		# 	return render(request,
+		#     'app/retrieve.html',
+		# 	{ 'user': request.user, 'error_message': 'weather' },
+		#     )
+		# request.session['wunderData'] = wunderData
+		# misrData = retrieve_misr_info(exifData['location'])
+		# if misrData is None:
+		# 	return render(request,
+		#     'app/retrieve.html',
+		# 	{ 'user': request.user, 'error_message': 'satellite' },
+		#     )
+		# request.session['misrData'] = misrData
+	    # return render(request,
+	    # 'app/retrieve.html',
+		# { 'user': request.user, 'exifData' : exifData,
+		# 'wunderData': wunderData, 'misrData': misrData, 'all_clear': True, },
+	    # )
+	    return render(request,
+	    'app/retrieve.html',
+		{ 'user': request.user, 'exifData' : 'exif here',
+		'wunderData': 'wunder here', 'misrData': 'misr here', 'all_clear': True,
+		'filename': filename, 'uploadedFileUrl': uploadedFileUrl,},
+	    )
+
+	return render(request,
+    'app/retrieve.html',
+	{ 'user': request.user, 'filename': filename, 'uploadedFileUrl': uploadedFileUrl,
+	'all_clear': False,},
+    )
+
+@login_required
+def results(request):
+	uploadedFileUrl = request.session['uploadedFileUrl']
+	filename = request.session['filename']
+	# exifData = request.session['exifData']
```

Comment:
All of the previously retrieved data is gathered from the session variable and used as input for the core algorithm, which returns aerosol data to be displayed. The functionality for adding the image and all data to our solr database has not yet been implemented